packages/frontend/src/components/UserSettings/UsersAndGroupsPanel/GroupsView.tsx

```diff
@@ -1,23 +1,123 @@
-import { Button, Stack, Table } from '@mantine/core';
-import { IconPlus } from '@tabler/icons-react';
-import { FC, useState } from 'react';
+import { Group as UserGroup } from '@lightdash/common';
+import {
+    Button,
+    Group,
+    Modal,
+    ModalProps,
+    Stack,
+    Table,
+    Text,
+    Title,
+} from '@mantine/core';
+import { IconAlertCircle, IconPlus, IconTrash } from '@tabler/icons-react';
+import { FC, useCallback, useState } from 'react';
 import { useTableStyles } from '../../../hooks/styles/useTableStyles';
-import { useOrganizationGroups } from '../../../hooks/useOrganizationGroups';
+import {
+    useGroupDeleteMutation,
+    useOrganizationGroups,
+} from '../../../hooks/useOrganizationGroups';
 import { useApp } from '../../../providers/AppProvider';
 import LoadingState from '../../common/LoadingState';
 import MantineIcon from '../../common/MantineIcon';
 import { SettingsCard } from '../../common/Settings/SettingsCard';
 import CreateGroupModal from './CreateGroupModal';
 
+const GroupListItem: FC<{
+    disabled?: boolean;
+    group: UserGroup;
+    onDelete: (group: UserGroup) => void;
+}> = ({ disabled, group, onDelete }) => {
+    return (
+        <tr>
+            <td width={500}>
+                <Text fz="xs" fw={400} color="gray.8">
+                    {group.name}
+                </Text>
+            </td>
+
+            <td>
+                <Text>No users</Text>
+            </td>
+            <td>
+                <Group position="right">
+                    <Button
+                        px="xs"
+                        variant="outline"
+                        onClick={() => onDelete(group)}
+                        disabled={disabled}
+                        color="red"
+                    >
+                        <MantineIcon icon={IconTrash} />
+                    </Button>
+                </Group>
+            </td>
+        </tr>
+    );
+};
+
+const DeleteGroupModal: FC<
+    ModalProps & {
+        group?: UserGroup;
+        disableControls?: boolean;
+        onAcceptDelete: () => void;
+    }
+> = ({ opened, group, disableControls, onAcceptDelete, onClose }) => {
+    return (
+        <Modal
+            opened={opened}
+            onClose={onClose}
+            title={
+                <Group spacing="xs">
+                    <MantineIcon size="lg" icon={IconAlertCircle} color="red" />
+                    <Title order={4}>Delete group {group?.name}</Title>
+                </Group>
+            }
+        >
+            <Text pb="md">Are you sure you want to delete this group?</Text>
+            <Group spacing="xs" position="right">
+                <Button
+                    disabled={disableControls}
+                    onClick={onClose}
+                    variant="outline"
+                    color="dark"
```

Comment:
Can we remove this? We don't do `color="dark"` for most `Cancel` buttons in Modals across the app. Unless I'm wrong!

Reply:
Keeping it this way for now as discussed in slack